### ISSUE TITLE
Bug: LeftOuter join panicked at src/stream/src/executor/managed_state/join/join_entry_state.rs:51:13

### ISSUE THREAD

**Issue by hzxa21** [risingwavelabs/risingwave#14040]
### Describe the bug

```
thread 'risingwave-streaming-actor' panicked at src/stream/src/executor/managed_state/join/join_entry_state.rs:51:13:
pk [...] should be in the cache
stack backtrace:
   0:     0x5593b72e3fc7 - std::backtrace_rs::backtrace::libunwind::trace::he8dc9e634c3d5b69
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x5593b72e3fc7 - std::backtrace_rs::backtrace::trace_unsynchronized::hcc1fc094cc2c1843
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x5593b72e3fc7 - std::sys_common::backtrace::_print_fmt::he6c8e9059e21630d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x5593b72e3fc7 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h145361bce15f4a6a
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x5593aea03a80 - core::fmt::rt::Argument::fmt::hb890552c7d6e8c2b
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/fmt/rt.rs:142:9
   5:     0x5593aea03a80 - core::fmt::write::h97d71643815ece8a
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/fmt/mod.rs:1117:17
   6:     0x5593b72dcbef - std::io::Write::write_fmt::h9145d12d2fb312aa
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/io/mod.rs:1762:15
   7:     0x5593b72e3da4 - std::sys_common::backtrace::_print::h34d8c7ca46068513
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x5593b72e3da4 - std::sys_common::backtrace::print::h69de4357a88b4f9d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x5593b72e5467 - std::panicking::default_hook::{{closure}}::hbf7d8b588847f0d6
  10:     0x5593b72e51c5 - std::panicking::default_hook::h797e21049b5f2e1d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:292:9
  11:     0x5593b6722b97 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h048b08ef2f029e65
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2021:9
  12:     0x5593b6722b97 - risingwave_rt::panic_hook::set_panic_hook::{{closure}}::h2a8bdac443388f10
                               at /risingwave/src/utils/runtime/src/panic_hook.rs:25:9
  13:     0x5593b6722b97 - std::panicking::update_hook::{{closure}}::h3589aef4f175aeb8
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:234:47
  14:     0x5593b72e5aba - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h49df07f7cb25728d
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/alloc/src/boxed.rs:2021:9
  15:     0x5593b72e5aba - std::panicking::rust_panic_with_hook::h4b1939882c095f9c
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:735:13
  16:     0x5593b72e583e - std::panicking::begin_panic_handler::{{closure}}::ha8983f1edca62350
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:609:13
  17:     0x5593b72e4486 - std::sys_common::backtrace::__rust_end_short_backtrace::h14cba478a9ee1fbd
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/sys_common/backtrace.rs:170:18
  18:     0x5593b72e55b2 - rust_begin_unwind
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/std/src/panicking.rs:597:5
  19:     0x5593aea009e5 - core::panicking::panic_fmt::h6433adef86579fb6
                               at /rustc/249624b5043013d18c00f0401ca431c1a6baa8cd/library/core/src/panicking.rs:72:14
  20:     0x5593b6e55904 - risingwave_stream::executor::managed_state::join::join_entry_state::JoinEntryState::remove::h0b81809130dd7096
                               at /risingwave/src/stream/src/executor/managed_state/join/join_entry_state.rs:51:13
  21:     0x5593b22dbe5c - risingwave_stream::executor::managed_state::join::JoinHashMap<K,S>::delete::h3504acbec91b1b65
                               at /risingwave/src/stream/src/executor/managed_state/join/mod.rs:549:13
  22:     0x5593b1214ff9 - risingwave_stream::executor::hash_join::HashJoinExecutor<K,S,_>::eq_join_oneside::{{closure}}::h24fa6736af757405
                               at /risingwave/src/stream/src/executor/hash_join.rs:1236:29
  23:     0x5593b1214ff9 - <futures_async_stream::try_stream::GenTryStream<G> as futures_core::stream::Stream>::poll_next::h689583320bd32219
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.9/src/lib.rs:506:33
  24:     0x5593b1214ff9 - risingwave_stream::executor::hash_join::HashJoinExecutor<K,S,_>::into_stream::{{closure}}::hc0c9796850e3f27e
                               at /risingwave/src/stream/src/executor/hash_join.rs:675:5
  25:     0x5593b0efda8b - <futures_async_stream::try_stream::GenTryStream<G> as futures_core::stream::Stream>::poll_next::h2de043d68292b7c7
                               at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/futures-async-stream-0.2.9/src/lib.rs:506:33
```

commit: 9ae8705451
Re-create the MV fixes the problem.

**Comment by fuyufjh** [comment]
Discussion #14031

**Comment by hzxa21** [comment]
potential fix: #15231
